pocket-android-core/src/main/java/network/pocket/core/model/Relay.kt

```diff
@@ -49,6 +32,9 @@ open class NewRelay(blockchain: String, netId: String, devId: String, data: Stri
      * @return whether it's correctly configured.
      */
     fun isValid(): Boolean {
-        return Utils.areDirty(this.blockchain, this.data, this.devId)
+        return when(Utils.areDirty(this.method, this.path)) {
```

Comment:
So here's the rational:
1. I changed the `areDirty` to return true if one of the args is null or empty.
2. There are 2 validation paths depending on what params they pass to the relay:
A. They pass path and method, which means the relay is valid when none of the following are empty or null: this.blockchain, this.netId, this.devId, this.method, this.path
B. They pass only the data param, which means the relay is valid when none of the following are empty or null: this.blockchain, this.netId, this.devId, this.data

This explains the 2 lines you don't understand. The when statement will jump to the `true` or `false` depending on whether or not the `path` and `method` params are dirty.

Reply:
Yeah but my problem is that they could pass a null `method` and it is still a valid REST request.

Reply:
What happens if they pass the data param and the method and path?

Reply:
1.- If they pass a null method it will jump into the following line: `true -> !Utils.areDirty(this.blockchain, this.netId, this.devId, this.data)` where the relay will be valid if they pass along `this.blockchain, this.netId, this.devId, this.data`.
2.- If they pass all 3 of them they would require these to be valid: `this.blockchain, this.netId, this.devId, this.method, this.path`

Reply:
1. - But it's possible to do a post REST in URL request and have data null correct?

Reply:
Correct, and that's contemplated in this validation scheme.

Reply:
> If they pass a null method it will jump into the following line: true -> !Utils.areDirty(this.blockchain, this.netId, this.devId, this.data) where the relay will be valid if they pass along this.blockchain, this.netId, this.devId, this.data.

So scenario: 
`null method AND null data = is invalid`

Reply:
Correct

Reply:
But that's not correct

Reply:
null method = POST

Reply:
Post with no data should be invalid, no? What I wanna avoid with this is having to rely on the server to error out, when we can easily catch it on the client side.

Reply:
> But it's possible to do a post REST in URL request and have data null correct?

> Correct, and that's contemplated in this validation scheme.

Yeah, I'm just saying with other SDKs this would be possible tho

Reply:
I see, so how do you suggest the validation go? We could check for blockchain, devid and netID only.

Reply:
Yeah I have validation in core hang on